static-analysis-plugin/plugin/src/main/groovy/com/novoda/staticanalysis/CheckstyleConfigurator.groovy

```diff
@@ -4,38 +4,48 @@ import groovy.util.slurpersupport.GPathResult
 import org.gradle.api.Project
 import org.gradle.api.Task
 import org.gradle.api.plugins.quality.Checkstyle
+import org.gradle.api.plugins.quality.CheckstyleExtension
 import org.gradle.internal.logging.ConsoleRenderer
 
 class CheckstyleConfigurator {
 
-    void configure(Project project, Violations violations, Task evaluateViolations) {
-        project.apply plugin: 'checkstyle'
-        project.checkstyle {
-            toolVersion = '7.1.2'
-        }
-        project.afterEvaluate {
-            boolean isAndroidApp = project.plugins.hasPlugin('com.android.application')
-            boolean isAndroidLib = project.plugins.hasPlugin('com.android.library')
-            if (isAndroidApp || isAndroidLib) {
-                def variants = isAndroidApp ? project.android.applicationVariants : project.android.libraryVariants
-                configureAndroid(project, variants)
+    void configure(Project project, Violations violations, StaticAnalysisExtension extension, Task evaluateViolations) {
```

Comment:
This method is becoming so messy that reviewing is very hard 😛 
It could help refactoring some blocks out of `configure`.